rest-e2e-tests/Makefile

```diff
@@ -0,0 +1,78 @@
+#
+# https://makefiletutorial.com
+# https://clarkgrubb.com/makefile-style-guide
+#
+
+# This Makefile uses .ONESHELL option supported by Make 3.82
+ifeq ($(filter oneshell,$(.FEATURES)),)
+$(error error: Your version of make ($(shell make -v|head -1|cut -f 3 -d ' ')) does not support .ONESHELL)
+endif
+
+LANG := en_US.UTF-8
+SHELL := bash
+.SHELLFLAGS := -eu -o pipefail -c # run '/bin/bash ... -c /bin/cmd'
+.ONESHELL:
+.DEFAULT_GOAL := all
+
+nvm_brew = /usr/local/opt/nvm/nvm.sh
+ifneq ("$(wildcard $(nvm_brew))", "")
+	nvm_sh = $(nvm_brew)
+endif
+nvm_default = $(HOME)/.nvm/nvm.sh
+ifneq ("$(wildcard $(nvm_default))", "")
+	nvm_sh = $(nvm_default)
+endif
+node_version = $(shell cat .nvmrc)
+define npm
+	@$(eval npm_args=$(1))
+	bash -l -c "source $(nvm_sh) && nvm exec $(node_version) npm $(npm_args)"
+endef
+
+all: clean vendor test/server/wait test ## Set up and test
+
+node_modules: ## Run 'npm ci' if directory doesn't exist
+	$(call npm, ci)
+
+vendor: node_modules ## Install source code packages
+
+engine_status_check_url := http://localhost:8081/streamr-core/api/v1/users/me
+.PHONY: test/server/wait
+test/server/wait: ## Wait for server to come online
+	export i=1
+	while true; do
+		http_code=$$(curl --silent --max-time 1 --output /dev/null --write-out "%{http_code}" $(engine_status_check_url)) || true
+		if [ "$$http_code" -eq 401 ]; then
+			echo "test/server/wait: $$(date -u) engine up and running"
+			break
+		else
+			echo "test/server/wait: $$(date -u) engine responding with HTTP status code $$http_code at $(engine_status_check_url)"
+			if [ "$$http_code" -eq 000 ]; then
+				sleep 3
+			fi
+		fi
+		i=$$(expr $$i + 1)
+		if [ $$i -eq 70 ]; then
+			echo "test/server/wait: timeout while waiting for engine to come online" 1>&2
+			exit 1
+		fi
+	done
+
+.PHONY: test
+test: vendor test/e2e test/stress ## Run tests
+
+.PHONY: test/e2e
```

Comment:
Why naming convention is different from main makefile where names are like `test-integration`?

Reply:
It's an experimental feature hidden behind the main Makefile. Let's see if those test target's get any use and if they are better the old way. Consider 'test/' as root 'folder' for all test related recipes and you can easily check what's available with `make test\t`.